week-4/Homework/mandatory/2-writers.js

```diff
@@ -1,44 +1,97 @@
-/* Challenge 1: Famous Writers
-Did you know you can also have an array of objects? We've created one for you here. Loop through the array, 
-and for each object, `console.log()` out the sentence:
+/* 
+Challenge 1: Famous Writers
 
-"Hi, my name is {firstName} {lastName}. I am {age} years old, and work as a {occupation}."
+Did you know you can also have an Array of Objects? You might think "This is madness!" but in everyday coding life
+it is quite frequent combination. Just think about it what benefits we can get from this construct.
 
-Here is the array:
+Object lets you store multiple values in a single variable, then you can store complex objects in an array.
+Let's assume you have list of data about people names and their birthday and you would like to print each name
+with corresponding birthdays together. Storing these information in different arrays and then pairing them up
+makes the iteration unnecessarily complicated, code will be less intuitive, needs extra cognitive effort to
+reason about and last but not least it can be error-prone for example you pick up the wrong birthday to a name.
 
+In this exercise you will practice how to access to Objects stored in Array and to their properties.
+You already know different ways how to loop through Arrays, it won't be different in this case too.
+The only extra is that you have to use values inside Objects.
 */
 
+// We've created an array of objects for you here:
 let writers = [
   {
     firstName: "Virginia",
     lastName: "Woolf",
     occupation: "writer",
     age: 59,
-    alive: false
+    alive: false,
   },
   {
     firstName: "Zadie",
     lastName: "Smith",
     occupation: "writer",
     age: 41,
-    alive: true
+    alive: true,
   },
   {
     firstName: "Jane",
     lastName: "Austen",
     occupation: "writer",
     age: 41,
-    alive: false
+    alive: false,
   },
   {
     firstName: "bell",
     lastName: "hooks",
     occupation: "writer",
     age: 64,
-    alive: true
-  }
+    alive: true,
+  },
 ];
 
 /*
-If you want an extra challenge, only `console.log()` the writers that are alive.
+Exercise 1:
+  Loop through the Array, and for each object, `console.log()` out the sentence and
+  insert corresponding values to the place holder that are indicated in courle braces:
+  "Hi, my name is {firstName} {lastName}. I am {age} years old, and work as a {occupation}."
+*/
+function introducing() {
+  for (let i = 0; i < writers.length; i++) {
+    console.log(
+      `Hi, my name is ${writers[i].firstName} ${writers[i].lastName}. I am ${writers[i].age} years old, and work as a ${writers[i].occupation}.`
+    );
+  }
+}
+introducing();

```

Comment:
Hi Altom, Your code is correct but you don't need to put the for loop inside a function as the for loop will print out the information on Writers.